### PR TITLE
New keysSet function for Data.VMap

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: libs/vector-map/src/Data/VMap.hs [hunk 1]
```diff
@@ -244,6 +247,10 @@ keys :: VG.Vector kv k => VMap kv vv k v -> [k]
 keys = VG.toList . keysVector . unVMap
 {-# INLINE keys #-}
 
+keysSet :: (Ord k, VG.Vector kv k) => VMap kv vv k v -> Set k
+keysSet = Set.fromList . keys
```
Comment (lehins) [inline libs/vector-map/src/Data/VMap.hs:251]: You can use `Set.fromDistinctAscList` here, because keys are unique and sorted.

**Comment by newhoggy** [comment]
Closing.  I ended up not using this new function.

**Review by lehins — CHANGES_REQUESTED** [review]
This seems like a useful function. One comment that will make it faster.

There are a couple of things I'd like to ask:

* A roundtrip test in libs/vector-map/test/Test/VMap.hs
* an entry in changelog

**Comment by lehins** [comment]
@newhoggy I left my review just in case you will change your mind on this PR :wink:

**Pull request closed without merging**